### PR TITLE
Bigtable: add CRUD for AppProfiles

### PR DESCRIPTION
**Pull request by igorbernstein2**

### CODE REVIEW

**Comment by igorbernstein2** [comment]
@kevin4508 can you take a look?

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/UpdateAppProfileRequest.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+  private UpdateAppProfileRequest(@Nonnull String instanceId, @Nonnull String appProfileId,
+      @Nonnull com.google.bigtable.admin.v2.UpdateAppProfileRequest.Builder builder) {
+    Preconditions.checkNotNull(instanceId, "instanceId must be set");
+    Preconditions.checkNotNull(appProfileId, "appProfileId must be set");
+    Preconditions.checkNotNull(builder, "builder must be set");
+
+    this.instanceId = instanceId;
+    this.appProfileId = appProfileId;
+    this.proto = builder;
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/UpdateAppProfileRequest.java]: "proto", "builder", "request", "requestBuilder"? pick one?

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java [hunk 2]
```diff
@@ -105,4 +127,332 @@ public ProjectName getProjectName() {
...
+   *
+   * <pre>{@code
+   *
+   * ApiFuture<AppProfile> existingAppProfileFuture = client.getAppProfileAsync("my-instance", "my-app-profile");
+   *
+   * ApiFuture<AppProfile> updatedAppProfileFuture = ApiFutures.transformAsync(
+   *   existingAppProfileFuture,
+   *   new ApiAsyncFunction<AppProfile, AppProfile>() {
+   *     @Override
+   *     public ApiFuture<AppProfile> apply(AppProfile existingAppProfile) {
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java]: nit: Perhaps add some update profile logic here before calling updateAppProfileAsync()?

**Comment by igorbernstein2** [comment]
All feedback addressed. @pongad please merge if this looks good to you

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/UpdateAppProfileRequest.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+ *
+ * <pre>{@code
+ * AppProfile existingAppProfile = ...;
+ * UpdateAppProfileRequest appProfileRequest = UpdateAppProfileRequest.of(existingAppProfile)
+ *   .setRoutingPolicy(SingleClusterRoutingPolicy.of("my-cluster"));
+ * }</pre>
+ *
+ * @see AppProfile for more details
+ */
+public final class UpdateAppProfileRequest {
```
Comment (pongad) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/UpdateAppProfileRequest.java]: I'd use AutoValue, but OK if you don't want to.

**Pull request merged**